### PR TITLE
update rank_zero condition for logging model summary

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Addresses a [comment](https://github.com/PyTorchLightning/pytorch-lightning/pull/9447#discussion_r706651980) by @ananthsub. 

Part of #9043 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

changelog not necessary, this callback is unreleased.

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/callbacks/model_summary.py [hunk 1]
```diff
@@ -50,14 +50,16 @@ def __init__(self, max_depth: Optional[int] = 1):
         self._max_depth: int = max_depth
 
     def on_pretrain_routine_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
-        if trainer.is_global_zero and self._max_depth is not None and not trainer.testing:
```
Comment (awaelchli) [inline pytorch_lightning/callbacks/model_summary.py:53]: reviewer: the check for testing gets removed here. it is redundant as we are in the "on_pretrain_routine_start" hook

File: pytorch_lightning/callbacks/model_summary.py [hunk 2]
```diff
@@ -50,14 +50,16 @@ def __init__(self, max_depth: Optional[int] = 1):
...
-            model_summary = summarize(pl_module, max_depth=self._max_depth)
+        if not self._max_depth:
+            return None
```
Comment (ananthsub) [inline pytorch_lightning/callbacks/model_summary.py:54]: nit
```suggestion
            return
```

Comment (awaelchli) [reply pytorch_lightning/callbacks/model_summary.py:54]: mypy strictly requires this :)

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/callbacks/model_summary.py (hunk 2 above)
Comment (ananthsub) [reply pytorch_lightning/callbacks/model_summary.py:54]: ah that's annoying :/ I haven't seen this when using pyre

**Review by ananthsub — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Pull request merged**